packages/backend/src/queryBuilder.ts

```diff
@@ -128,6 +131,105 @@ const getJoinType = (type: DbtModelJoinType = 'left') => {
     }
 };
 
+export const getCustomDimensionSql = ({
```

Comment:
Potential things to refactor:

I extracted all the logic for custom dimensions on a method, this way is easier to generate the sql, we should probably do the same for table calculations, etc.